core/trino-main/src/main/java/io/trino/operator/scalar/JsonStringArrayExtractScalar.java

```diff
@@ -0,0 +1,159 @@
+    private static final MethodHandle METHOD_HANDLE = methodHandle(JsonStringArrayExtractScalar.class, "extract", ArrayType.class, ArrayBlockBuilderAppender.class, ConnectorSession.class, Slice.class, JsonPath.class);
+
+    private static final JsonFactory JSON_FACTORY = createJsonFactory();
+
+    static {
+        // Changes factory. Necessary for JsonParser.readValueAsTree to work.
+        new ObjectMapper(JSON_FACTORY);
+    }
+
+    private JsonStringArrayExtractScalar()
+    {
+        super(FunctionMetadata.scalarBuilder(JSON_STRING_ARRAY_EXTRACT_SCALAR_NAME)
+                .signature(Signature.builder()
+                        .argumentType(new TypeSignature("varchar", typeVariable("N")))
+                        .longVariable("N")
+                        .argumentType(new TypeSignature(JsonPathType.NAME))
+                        .returnType(arrayType(new TypeSignature("varchar", typeVariable("N"))))
+                        .build())
+                .nullable()
+                .hidden()
+                .noDescription()
+                .build());
+    }
+
+    @Override
+    protected SpecializedSqlScalarFunction specialize(BoundSignature boundSignature)
+    {
+        checkArgument(boundSignature.getArity() == 2, "Expected arity to be 2");
+        ArrayType arrayType = (ArrayType) boundSignature.getReturnType();
+        ArrayBlockBuilderAppender arrayAppender = new ArrayBlockBuilderAppender(new VarcharBlockBuilderAppender(arrayType.getElementType()));
+        MethodHandle methodHandle = METHOD_HANDLE.bindTo(arrayType).bindTo(arrayAppender);
+
+        return new ChoicesSpecializedSqlScalarFunction(
+                boundSignature,
+                NULLABLE_RETURN,
+                ImmutableList.of(NEVER_NULL, NEVER_NULL),
+                methodHandle);
+    }
+
+    @UsedByGeneratedCode
+    public static Block extract(ArrayType arrayType, ArrayBlockBuilderAppender arrayAppender, ConnectorSession connectorSession, Slice json, JsonPath jsonPath)
+    {
+        try (JsonParser jsonParser = createJsonParser(JSON_FACTORY, json)) {
+            jsonParser.nextToken();
+            if (jsonParser.getCurrentToken() == JsonToken.VALUE_NULL) {
+                return null;
+            }
+            BlockBuilder blockBuilder = arrayType.createBlockBuilder(null, 1);
+            arrayAppender.append(jsonParser, jsonPath, blockBuilder);
+            Block block = blockBuilder.build();
+            return arrayType.getObject(block, 0);
+        }
+        catch (TrinoException | JsonCastException e) {
+            throw new TrinoException(INVALID_CAST_ARGUMENT, format("Cannot cast to %s. %s\n%s", arrayType, e.getMessage(), truncateIfNecessaryForErrorMessage(json)), e);
+        }
+        catch (Exception e) {
+            throw new TrinoException(INVALID_CAST_ARGUMENT, format("Cannot cast to %s.\n%s", arrayType, truncateIfNecessaryForErrorMessage(json)), e);
+        }
+    }
+
+    record VarcharBlockBuilderAppender(Type type)
```

Comment:
I don't think this record is needed. Can be static method method in `JsonStringArrayExtractScalar`